### PR TITLE
Fix appender code that might not initialize memory that might point at huge allocations

### PR DESCRIPTION
**Pull request by schveiguy**
Not going to reproduce all the junk I did in the bug report. See the report.

### CODE REVIEW

**Comment by schveiguy** [comment]
switching to draft, need to add a test.

**Comment by schveiguy** [comment]
well, I guess there isn't any sane testing harnesses I can incorporate here, and I don't want to add one for this. So it will have to go in with existing tests.

**Comment by jmdavis** [comment]
> well, I guess there isn't any sane testing harnesses I can incorporate here, and I don't want to add one for this. So it will have to go in with existing tests.

Wouldn't a test to make sure that the elements were zeroed out suffice?

**Comment by LightBender** [comment]
I'm inclined to agree with @jmdavis. Given the fragile nature of the proof-test and that it requires a whole-program testing framework that does not currently exist in Phobos, I would think that a test that ensures that the desired condition is met should be sufficient for our purposes. Such a test would prevent regressions which is mostly what we're after.

**Comment by schveiguy** [comment]
I can try and add a unittest that attempts to cause the problem to happen. Let's see if I can get it to fail with the original code though.

File: std/array.d [hunk 1]
```diff
@@ -4011,6 +4031,43 @@ if (isDynamicArray!A)
     app2.toString();
 }
 
+// https://issues.dlang.org/show_bug.cgi?id=24856
+@system unittest
+{
+    import core.memory : GC;
+    import std.stdio : writeln;
+    import std.algorithm : canFind;
```
Comment (thewilsonator) [inline std/array.d:4039]:
```
Check for package wide std.algorithm imports
grep -nr 'import std.algorithm : ' $(find etc std -name '*.d') ; test $? -eq 1
std/array.d:4039:    import std.algorithm : canFind;
```
I think it wants `import std.algorithm.searching : canFind;`

**Review by thewilsonator — APPROVED** [review]
otherwise looks good

File: std/array.d (hunk 1 above)
Comment (schveiguy) [reply std/array.d:4039]: oh how quaint...

**Pull request merged**